docs/user-guides/configuring-kepler.md

```diff
@@ -0,0 +1,255 @@
+# Configuring Kepler with Custom ConfigMaps
+
+This guide explains how to customize Kepler's configuration using the `additionalConfigMaps` feature in the PowerMonitor custom resource.
+
+## Overview
+
+The Kepler Operator allows you to customize Kepler's behavior by providing additional configuration through Kubernetes ConfigMaps. When you reference ConfigMaps in your PowerMonitor CR, the operator automatically:
+
+1. Merges your custom configuration with the default Kepler configuration
+2. Updates the Kepler DaemonSet with the new configuration
+3. Automatically triggers a rollout when the ConfigMap changes
+
+## Prerequisites
+
+- Kepler Operator installed in your cluster
+- PowerMonitor CR deployed
+- Access to create ConfigMaps in the namespace where Kepler is deployed (default: `power-monitor`)
+
+## Step-by-Step Configuration
+
+### Step 1: Create a ConfigMap with Your Custom Configuration
+
+Create a ConfigMap containing your Kepler configuration in a file named `config.yaml`. The ConfigMap must exist in the same namespace as the PowerMonitor deployment.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-kepler-config
+  namespace: power-monitor
+data:
+  config.yaml: |
+    metrics:
+      export:
+        interval: 10s
+    dev:
+      fake-cpu-meter:
+        enabled: false
+```
+
+Apply the ConfigMap:
+
+```bash
+kubectl apply -f my-kepler-config.yaml
+```
+
+### Step 2: Reference the ConfigMap in Your PowerMonitor CR
+
+Update your PowerMonitor custom resource to reference the ConfigMap you created:
+
+```yaml
+apiVersion: kepler.system.sustainable.computing.io/v1alpha1
+kind: PowerMonitor
+metadata:
+  name: power-monitor
+spec:
+  kepler:
+    config:
+      logLevel: info
+      additionalConfigMaps:
+        - name: my-kepler-config
+    deployment:
+      security:
+        mode: none
+```
+
+Apply the updated PowerMonitor CR:
+
+```bash
+kubectl apply -f power-monitor.yaml
+```
+
+### Step 3: Verify the Configuration
+
+The operator will automatically reconcile the DaemonSet with your new configuration. You can monitor the rollout:
+
+```bash
+# Check PowerMonitor status
+kubectl get powermonitor power-monitor
+
+# Watch DaemonSet rollout
+kubectl rollout status daemonset/power-monitor -n power-monitor
+
+# Verify the ConfigMap is mounted
+kubectl describe daemonset power-monitor -n power-monitor
+```
+
+## Configuration Examples
+
+### Example 1: Adjusting Monitoring Interval and Resource Tracking
+
+Customize how often Kepler monitors resources and how many terminated workloads to track:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kepler-monitor-config
+  namespace: power-monitor
+data:
+  config.yaml: |
+    monitor:
+      interval: 10s        # Monitor resources every 10 seconds (default: 5s)
```

Comment:
these can't overridden by additionalConfig since `spec.kepler.config.interval` should take precedence over any additional config.